internal/caller/tunnel.go

```diff
@@ -0,0 +1,69 @@
+package caller
+
+import (
+	"bytes"
+	"text/template"
+)
+
+var (
+	tunnelTemplate = `[Interface]
+PrivateKey = {{.privateKey}}
+Address = {{.address}}
+
+[Peer]
+PublicKey = {{.publicKey}}
+AllowedIPs = {{.allowedIPs}}
+Endpoint = {{.endpoint}}`
+)
+
+// Tunnel type represents a WireGuard tunnel configuration.
+type Tunnel struct {
+	privateKey string
+	address    string
+	publicKey  string
```

Comment:
public key, allowed IPs, and endpoint all define a peer.  Please split this out so that you can have a slice of TunnelPeer.  Alternatively this entire type you're building here is a type from wgctrl, it might be better to embed that type, and implement a method that converts it to a string.

Ultimately the problem is that this is moving ahead without the telephone, so I don't know yet what type the telephone will hand back.  This may lead to a lot of type shuffling.  In the ideal case, a tunnel config is just a result of what the telephone hands back.  If you move this type to internal/models, and call it an InterfaceConfig which encapsulates a slice of InterfacePeer, I can do all the marshaling internally there and hand it back to you directly with the fields filled in.

We can hop on a call and discuss this as well if this is unclear.